portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,30 +58,60 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         if(comments.size() < numComments){
           long id = entity.getKey().getId();
           String text = (String) entity.getProperty("text");
-          long timestamp = (long) entity.getProperty("timestamp");
 
-          Comment comment = new Comment(id,text,timestamp);
+          // Do the translation.
+          Translate translate = TranslateOptions.getDefaultInstance().getService();
+          Translation translation =
+            translate.translate(text, Translate.TranslateOption.targetLanguage(language));
+          String translatedText = translation.getTranslatedText();
+          
+          long timestamp = (long) entity.getProperty("timestamp");
+          String email = (String) entity.getProperty("email");
+          String nickname= (String) entity.getProperty("nickname");
+          Comment comment = new Comment(id,translatedText,timestamp,email, nickname);
           comments.add(comment);
         }
         else{
             break;
         } 
     }
     Gson gson = new Gson();
-    response.setContentType("application/json;");
+    response.setContentType("application/json");
+    response.setCharacterEncoding("UTF-8");
     response.getWriter().println(gson.toJson(comments));
+    
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String text = request.getParameter("text");
     long timestamp = System.currentTimeMillis();
+    String email = userService.getCurrentUser().getEmail();
+    String nickname = getUserNickname(userService.getCurrentUser().getUserId());
+ 
 
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("text",text);
     commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("email",email);
+    commentEntity.setProperty("nickname",nickname);
```

Comment:
The `getUserNickname` may return null and you could be setting a null value here. does the `.setProperty` function allow null values?

Reply:
I think they just come out as undefined.